cmake/OpenCVPCHSupport.cmake

```diff
@@ -56,6 +56,10 @@ MACRO(_PCH_GET_COMPILE_FLAGS _out_compile_flags)
           endforeach()
         endif()
 
+        GET_TARGET_PROPERTY(_cxx_standard ${_PCH_current_target} CXX_STANDARD)
+        if (_cxx_standard)
+            LIST(APPEND ${_out_compile_flags} "${CMAKE_CXX${_cxx_standard}_EXTENSION_COMPILE_OPTION}")
```

Comment:
> _EXTENSION

Why not `_STANDARD`?

> CMAKE_CXX11_EXTENSION_COMPILE_OPTION=-std=gnu++11
> CMAKE_CXX11_STANDARD_COMPILE_OPTION=-std=c++11

Reply:
I haven't checked cmake source when it uses `EXTENSION`, and when `STANDARD`.  On the g++ command line its `-std=gnu++11` this is why I choosed `EXTENSION`.

Reply:
CMake's logic is [here](https://gitlab.kitware.com/cmake/cmake/blob/b4dfe1d8163055f9f40a4c18f5eccccb28f06559/Source/cmLocalGenerator.cxx#L1527-1557)

Reply:
Updated the PR, now I check `CXX_EXTENSIONS` property to select the correct variable. On build command line I use `-DCMAKE_CXX_STANDARD=11`, together with `-DENABLE_CXX11=1`. It turns `CMAKE_CXX_EXTENSIONS=OFF` (cmake's default is `ON` and complains that `__STRICT_ANSI__` not defined)